oauth2/oauth2-client/src/main/java/org/springframework/security/oauth2/client/registration/ClientRegistration.java

```diff
@@ -794,6 +790,7 @@ public static final class Builder {
 			private boolean requireProofKey;
 
 			private Builder() {
+				this.requireProofKey = true;
```

Comment:
Please move this to variable initialization `private boolean requireProofKey = true;`